blog/_posts/2019-01-18-fluxdiffeq.md

```diff
@@ -0,0 +1,823 @@
+when we have more rabbits. The simplest way of encoding that is
+
+$$\text{rabbits}'(t) = \alpha\cdot \text{rabbits}(t)$$
+
+where $\alpha$ is some learnable constant. If you know your calculus, the solution
+here is exponential growth from the starting point with a growth rate $\alpha$:
+$\text{rabbits}(t_\text{start})e^(\alpha t)$. But notice that we didn't need to know the
+solution to the differential equation to validate the idea: we encoded the
+structure of the model and mathematics itself then outputs what the solution
+should be. Because of this, differential equations have been the tool of choice
+in most science. For example, physical laws tell you how electrical quantities
+emit forces ([Maxwell's Equations](https://en.wikipedia.org/wiki/Maxwell%27s_equations)).
+These are essentially equations of how things change and thus
+"where things will be" is the solution to a differential equation. But in recent
+decades this application has gone much further, with fields like systems biology
+learning about cellular interactions by encoding known biological structures and
+mathematically enumerating our assumptions or in targeted drug dosage through
+PK/PD modelling in systems pharmacology.
+
+So as our machine learning models grow and are hungry for larger and larger
+amounts of data, differential equations have become an attractive option for
+specifying nonlinearities in a learnable (via the parameters) but constrained
+form. They are essentially a way of incorporating prior domain-specific knowledge of the structural relations
+between the inputs and outputs. Given this way of looking at the two, both methods
+trade off advantages and disadvantages, making them complementary tools for modeling.
+It seems like a clear next step in scientific practice to start putting them
+together in new and exciting ways!
+
+## What is the Neural Ordinary Differential Equation (ODE)?
+
+The neural ordinary differential equation is one of many ways to put these two
+subjects together. The simplest way of explaining it is that, instead of
+learning the nonlinear transformation directly, we wish to learn the structures
+of the nonlinear transformation. Thus instead of doing $y=ML(x)$, we put the
+machine learning model on the derivative, $y'(x) = ML(x)$, and now solve the ODE.
+Why would you ever do this? Well, one motivation is that, if you were to do this
+and then solve the ODE using the simplest and most error prone method, the
+Euler method, what you get is equivalent to a [residual neural network](https://arxiv.org/abs/1512.03385).
+The way the Euler method works is based on the fact that $y'(x) = \frac{dy}{dx}$, thus
+
+$$\Delta y = (y_\text{next} - y_\text{prev}) = \Delta x\cdot ML(x) \text{ which implies that } y_{i+1} = y_{i} + \Delta x\cdot ML(x_{i}).$$
+
+Looking familiar now? This is the basis of a residual network (ResNet) which is a type of recurrent neural network (RNN),

```

Comment:
TO be fair I was working on off some a definition someone else showed me, and they may havebeen wrong or I may have misunderstood.

Reply:
As far as I know a ResNet is strictly feedforward and claiming that it is in some way recurrent would need a convincing motivation. As written it just risks the credibility of the post.

Reply:
The paper mentions RNNs but it takes a fairly "DL theory" perspective on what that means. I agree that we should leave it out and focus on the ResNet comparison

Reply:
If I remember correctly it argued that differential equations could model both ResNet and RNN but not that those two would be otherwise related. I also think focusing on ResNet is a good plan.